platform-tests/src/test/java/org/junit/platform/launcher/core/StoreSharingTests.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2015-2025 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.platform.launcher.core;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import org.junit.jupiter.api.Test;
+import org.junit.platform.engine.ExecutionRequest;
+import org.junit.platform.engine.support.store.Namespace;
+import org.junit.platform.fakes.TestEngineSpy;
+import org.junit.platform.fakes.TestEngineStub;
+
+/**
+ * @since 5.13
+ */
+class StoreSharingTests {
+
+	@Test
+	void twoDummyEnginesUseRequestLevelStore() {
+		TestEngineSpy engineWriter = new TestEngineSpy("Writer") {
+			@Override
+			public void execute(ExecutionRequest request) {
+				request.getRequestLevelStore().put(Namespace.GLOBAL, "sharedKey", "Hello from Writer");
+				super.execute(request);
+			}
+		};
+
+		TestEngineStub engineReader = new TestEngineStub("Reader") {
+			@Override
+			public void execute(ExecutionRequest request) {
+				Object value = request.getRequestLevelStore().get(Namespace.GLOBAL, "sharedKey");
+				assertEquals("Hello from Writer", value);
+				super.execute(request);
+			}
+		};
+
+		ExecutionRequest request = mock(ExecutionRequest.class);
+		when(request.getRequestLevelStore()).thenReturn(
+			NamespacedHierarchicalStoreProviders.dummyNamespacedHierarchicalStore());
+
+		engineWriter.execute(request);
+		engineReader.execute(request);
```

Comment:
I think we should go through `Launcher.execute` here rather than calling `TestEngine.execute` directly. That way, we would also test the `Launcher`'s handling of the store meets our expectations.